src/sentry/rules/conditions/tagged_event.py

```diff
@@ -93,6 +128,24 @@ def passes(self, event, state, **kwargs):
             if k.lower() == key or tagstore.get_standardized_key(k) == key
         )
 
+        # specific case to handle if the user wants to match 'release equals latest'
+        if key == "release" and value == "latest":
+            latest_release = self.get_latest_release(event)
+            if not latest_release:
+                return False
+
+            if match == MatchType.EQUAL:
+                ret_value = True
+            elif match == MatchType.NOT_EQUAL:
+                ret_value = False
+            else:
+                return False
+
+            for t_value in values:
+                if t_value == latest_release.version:
+                    return ret_value
```

Comment:
Should we just set `value = latest_release.version` here and then let the match logic later on handle it? We could still validate that `match` is `EQUAL`/`NOT_EQUAL` here too if we need it.